src/pybind/mgr/nfs/export/export_utils.py

```diff
@@ -0,0 +1,322 @@
+class GaneshaConfParser:
+    def __init__(self, raw_config):
+        self.pos = 0
+        self.text = ""
+        self.clean_config(raw_config)
+
+    def clean_config(self, raw_config):
+        for line in raw_config.split("\n"):
+            self.text += line
+            if line.startswith("%"):
+                self.text += "\n"
+
+    def remove_whitespaces_quotes(self):
+        if self.text.startswith("%url"):
+            self.text = self.text.replace('"', "")
+        else:
+            self.text = "".join(self.text.split())
+
+    def stream(self):
+        return self.text[self.pos:]
+
+    def parse_block_name(self):
```

Comment:
Did you consider reusing some existing pseudo-'parser' like `shlex`? It doesn't fully resolve the parsing but at least it can simplify the process (removes # comments, newlines, whitespace, etc.) and allows you to focus on just processing the `{...}` blocks:
```py
>>> from shlex import shlex

>> config = 'EXPORT{\n    Export_Id = 1 ;   # Export ID unique to each export\n    Path = "volume_path";  # Path of the volume to be exported. Eg: "/test_volume"\n\n    FSAL {\n        name = GLUSTER;\n        hostname = "10.xx.xx.xx";  # IP of one of the nodes in the trusted pool\n        volume = "volume_name";  # Volume name. Eg: "test_volume"\n    }\n\n    Access_type = RW;    # Access permissions\n    Squash = No_root_squash; # To enable/disable root squashing\n    Disable_ACL = TRUE;  # To enable/disable ACL\n    Pseudo = "pseudo_path";  # NFSv4 pseudo path for this export. Eg: "/test_volume_pseudo"\n    Protocols = "3","4" ;    # NFS protocols supported\n    Transports = "UDP","TCP" ; # Transport protocols supported\n    SecType = "sys";     # Security flavors supported\n}\n'

>>> list(shlex.shlex(config, posix=True))
['EXPORT', '{', 'Export_Id', '=', '1', ';', 'Path', '=', 'volume_path', ';', 'FSAL', '{', 'name', '=', 'GLUSTER', ';', 'hostname', '=', '10.xx.xx.xx', ';', 'volume', '=', 'volume_name', ';', '}', 'Access_type', '=', 'RW', ';', 'Squash', '=', 'No_root_squash', ';', 'Disable_ACL', '=', 'TRUE', ';', 'Pseudo', '=', 'pseudo_path', ';', 'Protocols', '=', '3', ',', '4', ';', 'Transports', '=', 'UDP', ',', 'TCP', ';', 'SecType', '=', 'sys', ';', '}']

Reply:
No, I will look into this in a new PR.